### PR TITLE
Kerberos configurable && template simplification

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Fixes for way kerberos keytabs and principals for each service are passed to ansible. Before principals and keytab file names had to be set, now they can be configurable.

Also major simplification of the .properties file jinja templates. Notice the if statements in the templates. Now there is logic around why properties are set in the templates.

Also including prometheus exporter installation/ configuration for kafka broker

Fixes # (issue)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested installing all security modes and then ran system test

**Test Configuration**:

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: roles/confluent.kerberos/defaults/main.yml [hunk 1]
```diff
@@ -1,20 +1,19 @@
 kerberos:
   redhat_packages:
-    - cyrus-sasl-gssapi
-    - libselinux-python
+    # - cyrus-sasl-gssapi
+    # - libselinux-python
```
Comment (domenicbove) [inline roles/confluent.kerberos/defaults/main.yml]: remove commented code

File: roles/confluent.schema_registry/templates/schema-registry.properties.j2 [hunk 2]
```diff
@@ -1,4 +1,30 @@
 # Maintained by Ansible
-kafkastore.bootstrap.servers={% for host in groups['kafka_broker'] %}{% if loop.index > 1%},{% endif %}PLAINTEXT://{{ host }}:{{kafka_port}}{% endfor %}
+{% for key, value in schema_registry.properties.items() %}
+{{key}}={{value}}
+{% endfor %}
 
-{% include './includes/base_schema-registry.properties.j2' %}
```
Comment (domenicbove) [inline roles/confluent.schema_registry/templates/schema-registry.properties.j2]: what happened to the schema_registry prop

File: roles/confluent.zookeeper/defaults/main.yml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 zookeeper_open_file_limit: "{{open_file_limit}}"
 
-zookeeper_jaas_file: /etc/kafka/kafka_server_jaas.conf
+# zookeeper_jaas_file: /etc/kafka/kafka_server_jaas.conf
```
Comment (domenicbove) [inline roles/confluent.zookeeper/defaults/main.yml]: remove commented code

File: ansible.cfg [hunk 4]
```diff
@@ -1,2 +1,5 @@
 [defaults]
 hash_behaviour=merge
+
+[ssh_connection]
+pipelining = True
```
Comment (JumaX) [inline ansible.cfg]: This doesn't look related to the kerberos changes.  It addresses another request we had regarding performance.  Are there any other changes to be aware of?

File: roles/confluent.common/tasks/redhat.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 - name: Install Java
   yum:
     name: "{{redhat_java_package_name}}"
-    state: latest
+    state: present
```
Comment (JumaX) [inline roles/confluent.common/tasks/redhat.yml]: This appears to be a correction for a different issue, can you confirm this?  As its not mentioned in the description.

File: roles/confluent.control_center/tasks/main.yml [hunk 6]
```diff
@@ -14,89 +14,43 @@
...
-    group: "{{control_center.group}}"
-  when: security_mode == "plaintext"
-  notify:
-    - restart control center
-- name: control center create config
-  template:
-    src: control-center_ssl.properties.j2
-    dest: "{{control_center.config_file}}"
-    mode: 0640
-    owner: "{{control_center.user}}"
```
Comment (JumaX) [inline roles/confluent.control_center/tasks/main.yml]: The request is to modify the way kerberos security is handled.  Why are we refactoring how all templates work?

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 7]
```diff
@@ -13,108 +13,64 @@
...
-    group: "{{kafka_broker.group}}"
-  when: security_mode == "sasl_ssl"
-  notify:
-    - restart kafka
-- name: broker ssl custom signed config
-  template:
-    src: server_ssl_self_signed.properties.j2
-    dest: "{{kafka_broker.config_file}}"
-    mode: 0640
-    owner: "{{kafka_broker.user}}"
```
Comment (JumaX) [inline roles/confluent.kafka_broker/tasks/main.yml]: Why are refactoring how all of the templates work?  This PR should just be updating Kerberos to use better encryption types, and make the krb5.conf configurable.

File: roles/confluent.schema_registry/tasks/main.yml [hunk 8]
```diff
@@ -14,89 +14,41 @@
...
-    src: schema-registry_sasl_ssl.properties.j2
-    dest: "{{schema_registry.config_file}}"
-    mode: 0640
-    owner: "{{schema_registry.user}}"
-    group: "{{schema_registry.group}}"
-  when: security_mode == "sasl_ssl"
-  notify:
-    - restart schema-registry
-- name: schema-registry create config
-  template:
```
Comment (JumaX) [inline roles/confluent.schema_registry/tasks/main.yml]: See previous comment regarding refactoring.

File: roles/confluent.ksql/tasks/main.yml [hunk 9]
```diff
@@ -14,106 +14,62 @@
...
-    src: ksql_ssl.properties.j2
-    dest: "{{ksql.config_file}}"
-    mode: 0640
-    owner: "{{ksql.user}}"
-    group: "{{ksql.group}}"
-  when: security_mode == "ssl"
-  notify:
-    - restart ksql
-- name: ksql create config
-  template:
```
Comment (JumaX) [inline roles/confluent.ksql/tasks/main.yml]: see previous comment regarding refactoring.

File: roles/confluent.kafka_connect/tasks/main.yml [hunk 10]
```diff
@@ -14,87 +14,43 @@
...
-  template:
-    src: connect-distributed_ssl.properties.j2
-    dest: "{{kafka_connect.config_file}}"
-    mode: 0640
-    owner: "{{kafka_connect.user}}"
-    group: "{{kafka_connect.group}}"
-  when: security_mode == "ssl"
-  notify:
-    - restart connect distributed
-- name: connect distributed create config
```
Comment (JumaX) [inline roles/confluent.kafka_connect/tasks/main.yml]: See previous comment about refactoring.

File: roles/confluent.control_center/tasks/main.yml (hunk 6 above)
Comment (domenicbove) [reply roles/confluent.control_center/tasks/main.yml]: template refactoring is in the description

File: roles/confluent.common/tasks/redhat.yml (hunk 5 above)
Comment (domenicbove) [reply roles/confluent.common/tasks/redhat.yml]: sorry this is not related, but I realized we dont really have a requirement that java 1.8 is the latest, but present. Its up to the users when they want to update their java

File: roles/confluent.kafka_broker/tasks/main.yml [hunk 11]
```diff
@@ -13,111 +13,73 @@
...
     dest: "{{kafka_broker_jaas_file}}"
     mode: 0640
     owner: "{{kafka_broker.user}}"
     group: "{{kafka_broker.group}}"
-  when: security_mode == "kerberos" or security_mode == "kerberos_ssl" or security_mode == "kerberos_ssl_customcerts"
+  when: sasl_protocol == "kerberos"
   notify:
     - restart kafka
 
+- name: deploy JMX exporter config file
```
Comment (domenicbove) [inline roles/confluent.kafka_broker/tasks/main.yml]: Capitalize?

File: roles/confluent.common/tasks/main.yml [hunk 12]
```diff
@@ -22,3 +22,16 @@
     url: "http://search.maven.org/remotecontent?filepath=org/jolokia/jolokia-jvm/{{jolokia_version}}/jolokia-jvm-{{jolokia_version}}-agent.jar"
     dest: "{{ jolokia_jar_path }}"
   when: jolokia_enabled|bool
+
+- name: set up prometheus install folder
```
Comment (domenicbove) [inline roles/confluent.common/tasks/main.yml]: capitalize
